### PR TITLE
ENH: Fixed the shifted column names in summary.to_latex()

### PR DESCRIPTION
**Pull request by bashtage**
Fix header alignment to prevent extra table

- [X] closes #6896 
- [X] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/iolib/summary2.py [hunk 1]
```diff
@@ -582,6 +582,7 @@ def _df_to_simpletable(df, align='r', float_format="%.4f", header=True,
     st = SimpleTable(np.array(dat), headers=headers, stubs=stubs,
                      ltx_fmt=fmt_latex, txt_fmt=fmt_txt)
     st.output_formats['latex']['data_aligns'] = align
+    st.output_formats['latex']['header_align'] = align
```
Comment (Jimmy2027) [inline statsmodels/iolib/summary2.py:585]: Does this line solve the problem of the shifted column names? I have tried it, but the problem described in #6896 remains

**Comment by bashtage** [comment]
It fixes is on master.

**Comment by bashtage** [comment]
Not the most beautiful but the headers are aligned.

image

**Comment by bashtage** [comment]
The extra hline is uggo though.

**Comment by Jimmy2027** [comment]
It works for me too in master, thanks! I proposed to remove the hline in #6905

**Pull request merged**